<html>
  <head>
      <meta charset="UTF-8"/>
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
<title>生信学渣</title>
<link rel="shortcut icon" href="https://wangxf133.github.io/favicon.ico?v=1578404663540">
<link rel="stylesheet" href="https://wangxf133.github.io/styles/main.css">
<meta name="description" content="学习 记录 成长" />

<link rel="stylesheet" href="/media/fonts/font-awesome.css">
<link href="//fonts.googleapis.com/css?family=Monda:300,300italic,400,400italic,700,700italic|Roboto Slab:300,300italic,400,400italic,700,700italic|Rosario:300,300italic,400,400italic,700,700italic|PT Mono:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">

<script src="https://cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/velocity-animate@1.5.0/velocity.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/velocity-animate@1.5.0/velocity.ui.min.js"></script>

<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.css" integrity="sha384-zB1R0rpPzHqg7Kpt0Aljp8JPLqbXI3bhnPWROx27a9N0Ll6ZP/+DiW/UqRcLbRjq" crossorigin="anonymous">
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/katex.min.js" integrity="sha384-y23I5Q6l+B6vatafAwxRu/0oK/79VlbSz7Q9aiSZUvyWYIYsd+qj+o24G5ZU2zJz" crossorigin="anonymous"></script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.11.1/dist/contrib/auto-render.min.js" integrity="sha384-kWPLUVMOks5AQFrykwIup5lo0m3iMkkHrD0uJ4H5cjeGihAutqP0yW0J6dpFiVkI" crossorigin="anonymous"
    onload="renderMathInElement(document.body);"></script>
  </head>
  <body>
    <div class="head-top-line"></div>
    <div class="header-box">
      
<header class="header">
  <div class="blog-header" id="header">
    <div class="site-meta">
      <div class="nav-toggle" id="nav_toggle">
        <div class="toggle-box">
          <div class="line line-top"></div>
          <div class="line line-center"></div>
          <div class="line line-bottom"></div>
        </div>
      </div>
      <div class="site-title">
        <a href="/" class="brand">
          <span>生信学渣</span>
        </a>
      </div>
      <p class="subtitle">Quick notes</p>
    </div>
    <nav class="site-nav" id="site_nav">
      <ul id="nav_ul">
        
          <li class="nav-item ">
            
              <a href="/">
                <i class="fa fa-home"></i> 首页
              </a>
            
          </li>
        
          <li class="nav-item ">
            
              <a href="/archives">
                <i class="fa fa-archive"></i> 归档
              </a>
            
          </li>
        
          <li class="nav-item ">
            
              <a href="/tags">
                <i class="fa fa-tags"></i> 标签
              </a>
            
          </li>
        
          <li class="nav-item ">
            
              <a href="https://wangxf133.github.io/post/guan-yu/">
                <i class="fa fa-"></i> 关于
              </a>
            
          </li>
        
          <li class="nav-item ">
            
              <a href="https://wangxf133.github.io/post/jianli/">
                <i class="fa fa-"></i> 简历
              </a>
            
          </li>
        
        <li class="nav-item">
          <a>
            <i class="fa fa-search"></i> 搜索
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
<script type="text/javascript"> 
 
  let showNav = true;

  let navToggle = document.querySelector('#nav_toggle'),
  siteNav = document.querySelector('#site_nav');
  
  function navClick() {
    let sideBar = document.querySelector('.sidebar');
    let navUl = document.querySelector('#nav_ul');
    navToggle.classList.toggle('nav-toggle-active');
    siteNav.classList.toggle('nav-menu-active');
    if (siteNav.classList.contains('nav-menu-active')) {
      siteNav.style = "height: " + (navUl.children.length * 42) +"px !important";
    } else {
      siteNav.style = "";
    }
  }

  navToggle.addEventListener('click',navClick);  
</script>
    </div>
    <div class="main-continer">
      <div class="section-layout">
        <div class="section-layout-wrapper">
          
<div class="sidebar">
    <div class="sidebar-wrapper" id="sidebar">
      
        <div class="post-list-sidebar">
          <div class="sidebar-title">
            <span id="tocSideBar" class="sidebar-title-item sidebar-title-active">文章目录</span>
            <span id="metaSideBar" class="sidebar-title-item">站点概览</span>
          </div>
        </div>
      
      <div class="sidebar-body" id="sidebar_body">
        
          <div class="post-side-meta" id="post_side_meta">
            <div class="sidebar-wrapper">
  <div class="sidebar-item">
    <img class="site-author-image" src="https://wangxf133.github.io/images/avatar.png"/>
    <p class="site-author-name">生信学渣</p>
    <p class="site-description">学习 记录 成长</p>
  </div>
  <div class="sidebar-item side-item-stat">
    <div class="sidebar-item-box">
      <a href="/archives/">
        
        <span class="site-item-stat-count">18</span>
        <span class="site-item-stat-name">文章</span>
      </a>
    </div>
    <div class="sidebar-item-box">
      <a href="">
        <span class="site-item-stat-count">25</span>
        <span class="site-item-stat-name">分类</span>
      </a>
    </div>
    <div class="sidebar-item-box">
      <a href="/tags/">
        <span class="site-item-stat-count">25</span>
        <span class="site-item-stat-name">标签</span>
      </a>
    </div>
  </div>
  <div class="sidebar-item">
    <span class="site-item-rss">
        <i class="fa fa-rss"></i>
        <a href="https://wangxf133.github.io/atom.xml" target="_blank">RSS</a>
    </span>
  </div>
  <div class="sidebar-item sidebar-item-social">
    <div class="social-item">
      <a href="">
        <i class="fa fa-github"></i> GitHub
      </a>
      <a href="">
        <i class="fa fa-twitter"></i> Twitter
      </a>
    </div>
    <div class="social-item">
      <a href="">
        <i class="fa fa-globe"></i> 豆瓣
      </a>
      <a href="">
        <i class="fa fa-globe"></i> 知乎
      </a>
    </div>
  </div>
</div>
          </div>
          <div class="post-toc sidebar-body-active" id="post_toc" style="opacity: 1;">
            <div class="toc-box">
  <div class="toc-wrapper" id="toc_wrapper">
    <ul class="markdownIt-TOC">
<li>
<ul>
<li><a href="#1-%E5%9B%BD%E9%99%85%E5%B0%8F%E9%BA%A6%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%B5%8B%E5%BA%8F%E8%81%94%E7%9B%9F">1. 国际小麦基因组测序联盟</a></li>
<li><a href="#2-iwgsc%E7%9A%84%E5%9B%9B%E4%B8%AA%E5%85%B3%E9%94%AE%E9%87%8C%E7%A8%8B%E7%A2%91">2. IWGSC的四个关键里程碑</a></li>
<li><a href="#3-%E5%9F%BA%E5%9B%A0%E7%BB%84%E6%B5%8B%E5%BA%8F%E6%9D%90%E6%96%99-%E4%B8%AD%E5%9B%BD%E6%98%A5%E5%93%81%E7%A7%8D">3. 基因组测序材料-中国春品种</a></li>
<li><a href="#4-%E6%96%87%E7%AB%A0%E8%A7%A3%E8%AF%BB">4. 文章解读</a></li>
<li><a href="#5-%E6%95%B0%E6%8D%AE%E8%B5%84%E6%BA%90">5. 数据资源</a></li>
<li><a href="#6-%E5%B0%8F%E9%BA%A6%E5%9F%BA%E5%9B%A0%E7%BB%84%E4%B8%BB%E8%A6%81%E5%B7%B2%E5%8F%91%E8%A1%A8%E6%96%87%E7%8C%AE">6. 小麦基因组主要已发表文献</a></li>
</ul>
</li>
</ul>

  </div>
</div>

<script>

let lastTop = 0, lList = [], hList = [], postBody, lastIndex = -1; 
let active = 'active-show', activeClass = 'active-current';
let tocWrapper = document.querySelector('#toc_wrapper');
let tocContent = tocWrapper.children[0];

function addTocNumber(elem, deep) {
  if (!elem) {
    return;
  }
  let prop = elem.__proto__;

  if (prop === HTMLUListElement.prototype) {
    for (let i = 0; i < elem.children.length; i++) {
      addTocNumber(elem.children[i], deep + (i + 1) + '.');
    }
  } else if (prop === HTMLLIElement.prototype) {
    // 保存li元素
    lList.push(elem);
    for (let i = 0; i < elem.children.length; i++) {
      let cur = elem.children[i];
      if (cur.__proto__ === HTMLAnchorElement.prototype) {
        cur.text =  deep + ' ' + cur.text;
      } else if (cur.__proto__ === HTMLUListElement.prototype) {
        addTocNumber(cur, deep);
      }
    }
  }
}


document.addEventListener('scroll', function(e) {
  if (lList.length <= 0) {
    return;
  }
  let scrollTop = document.body.scrollTop;
  let dir;

  if (lastTop - scrollTop > 0) {
    dir = 'up';
  } else {
    dir = 'down';
  }

  lastTop = scrollTop;
  if (scrollTop <= 0) {
    if (lastIndex >= 0 && lastIndex < hList.length) {
      lList[lastIndex].classList.remove(activeClass);
    }
    return;
  }

  let current = 0, hasFind = false;
  for (let i = 0; i < hList.length; i++) {
    if (hList[i].offsetTop > scrollTop) {
      current = i;
      hasFind = true;
      break;
    }
  }
  if (!hasFind && scrollTop > lList[lList.length - 1].offsetTop) {
    current = hList.length - 1;
  } else {
    current--;
  }
  if (dir === 'down') {
    if (current > lastIndex) {
      addActiveClass(current);
      removeActiveClass(lastIndex) 
      lastIndex = current;
      removeParentActiveClass();
      lList[current] && addActiveLiElemment(lList[current].parentElement,tocContent);
    }
  } else {
    if (current < lastIndex) {
      addActiveClass(current);
      removeActiveClass(lastIndex);
      lastIndex = current;
      removeParentActiveClass();
      lList[current] && addActiveLiElemment(lList[current].parentElement,tocContent);
    }
  }
});

function removeParentActiveClass() {
  let parents = tocContent.querySelectorAll('.'+active)
  parents.forEach(function(elem) {
    elem.classList.remove(active);
  });
}

function addActiveClass(index) {
  if (index >= 0 && index < hList.length) {
    lList[index].classList.add(activeClass);
  }
}

function removeActiveClass(index) {
  if (index >= 0 && index < hList.length) {
    lList[index].classList.remove(activeClass);
  }
}

function addActiveLiElemment(elem, parent) {
  if (!elem || elem === parent) {
    return;
  } else {
    if (elem.__proto__ === HTMLLIElement.prototype) {
      elem.classList.add(active);
    }
    addActiveLiElemment(elem.parentElement, parent);
  }
}

function showToc() {
  if (tocWrapper) {
    postBody = document.querySelector('#post_body');
    for (let i = 0; i < postBody.children.length; i++) {
      if (postBody.children[i].__proto__ === HTMLHeadingElement.prototype) {
        hList.push(postBody.children[i]);
      }
    }

    if (hList.length > 10) {
      active = 'active-hidden'
      tocContent.classList.add('closed');
    } else {
      tocContent.classList.add('expanded');
    }
  }
}
addTocNumber(tocContent, '');

window.addEventListener('load', function() {
  showToc();
  document.querySelector('#sidebar').style='display: block;';
  tocWrapper.classList.add('toc-active');
  setTimeout(function() {
    if ("createEvent" in document) {
      let evt = document.createEvent("HTMLEvents");
      evt.initEvent("scroll", false, true);
      document.dispatchEvent(evt);
    }
    else {
      document.fireEvent("scroll");
    }
  }, 500)
})

</script>
          </div>
        
      </div>
    </div>
</div>
<script>
  const SIDEBAR_TITLE_ACTIVE = 'sidebar-title-active';
  const SIDEBAR_BODY_ACTIVE = 'sidebar-body-active';
  const SLIDE_UP_IN = 'slide-up-in';

  let sidebar = document.querySelector('#sidebar'),
  tocSideBar = document.querySelector('#tocSideBar'),
  metaSideBar = document.querySelector('#metaSideBar'),
  postToc = document.querySelector('#post_toc'),
  postSiteMeta = document.querySelector('#post_side_meta'),
  sidebarTitle = document.querySelector('.sidebar-title'),
  sidebarBody = document.querySelector('#sidebar_body');

  tocSideBar.addEventListener('click', (e) => {
    toggleSidebar(e);
  });

  metaSideBar.addEventListener('click', (e) => {
    toggleSidebar(e);
  });

  function toggleSidebar(e) {
    let currentTitle = document.querySelector("."+SIDEBAR_TITLE_ACTIVE);
    if (currentTitle == e.srcElement) {
      return ;
    }
    let current, showElement, hideElement;
    if (e.srcElement == metaSideBar) {
      showElement = postSiteMeta;
      hideElement = postToc;
    } else if (e.srcElement == tocSideBar){
      showElement = postToc;
      hideElement = postSiteMeta;
    }
    currentTitle.classList.remove(SIDEBAR_TITLE_ACTIVE);
    e.srcElement.classList.add(SIDEBAR_TITLE_ACTIVE);

    window.Velocity(hideElement, 'stop');
    window.Velocity(hideElement, 'transition.slideUpOut', {
      display: 'none',
      duration: 200,
      complete: function () {
        window.Velocity(showElement, 'transition.slideDownIn', {
          duration: 200
        });
      }
    })
    hideElement.classList.remove(SIDEBAR_BODY_ACTIVE);
    showElement.classList.add(SIDEBAR_BODY_ACTIVE);

  }

  postToc.addEventListener('transitionend', function() {
    this.classList.remove(SLIDE_UP_IN);
  });

  let limitTop = document.querySelector('#nav_ul').children.length * 42 + 162;
  let hasFix = false;
  window.addEventListener('scroll', function(e) {
    if (document.body.scrollTop >= limitTop) {
      if (!hasFix) {
        sidebar.classList.add('sidebar-fixed');
        hasFix = true;
      }
    } else {
      if (hasFix) {
        sidebar.classList.remove('sidebar-fixed');
        hasFix = false;
      }
    }
  });
</script>
          <div class="section-box">
            <div class="section post">
              <div class="article-box">
    <header class="post-header">
  <h1 class="post-title">
    <a class="post-title-link" href="https://wangxf133.github.io/post/science-chong-bang-orli-shi-13-nian-liu-bei-ti-xiao-mai-ji-yin-zu-tu-pu-zhong-wan-cheng-xiang-xi-jie-du">
      Science重磅|历时13年六倍体小麦基因组图谱终完成（详细解读）
    </a>
  </h1>
  <div class="post-meta">
    <span class="meta-item pc-show">
      <i class="fa fa-calendar-o"></i>
      <span>发布于</span>
      <span>2019-03-27</span>
    </span>
    
      <span class="meta-item">
        <span class="post-meta-divider pc-show">|</span>
        <i class="fa fa-folder-o"></i>
        <span class="pc-show">分类于</span>
        
          
            <a href="https://wangxf133.github.io/tag/A3h6Jlq6p">
              <span>iwgsc</span>
            </a>、
          
        
          
            <a href="https://wangxf133.github.io/tag/We7V91z05">
              <span>基因组</span>
            </a>、
          
        
          
            <a href="https://wangxf133.github.io/tag/jOhsegpai">
              <span>小麦</span>
            </a>
          
        
      </span>
    
    <span class="meta-item">
      <span class="post-meta-divider">|</span>
      <i class="fa fa-clock-o"></i>
      <span>13分钟</span>
    </span>
    <span class="meta-item">
      <span class="post-meta-divider">|</span>
      <i class="fa fa-file-word-o"></i>
      <span>3310<span class="pc-show">字数</span></span>
    </span>
  </div>
</header>
</div>
              <div class="post-body next-md-body" id="post_body">
                <p>转自 生新人发表的<a href="https://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651160081&amp;idx=1&amp;sn=2bc94fcce9b3c939110448b99f710c3a&amp;chksm=8b5c797ebc2bf06847d02884e19b0281752a796dc8df1897c5f7cae78e0eeef25a36eb58bb1a&amp;mpshare=1&amp;scene=23&amp;srcid=0817my25QXeITXGASjxPCXwz#rd">Science重磅|历时13年六倍体小麦基因组图谱终完成（详细解读）</a></p>
<p>2018年8月17日Science杂志在线刊发题为“Shifting the limits in wheat research and breeding using a fully annotated reference genome”的研究论文，相比于2014年7月发布的草图序列，该版本序列组装指标明显提升，基因组序列由原先的10.2Gb上升到14.5Gb，Scaffold N50提升到22.8Mb。该研究论文是在国际小麦基因组测序联盟（International Wheat Genome Sequencing Consortium，IWGSC）协作下完成的，西北农林科技大学旱区作物逆境生物学国家重点实验室的宋卫宁教授团队作为中国唯一参与并承担实质性研究工作的团队，完成了其中7DL染色体物理图谱构建及序列图谱破译工作。</p>
 <!--more-->
<h2 id="1-国际小麦基因组测序联盟">1. 国际小麦基因组测序联盟</h2>
<p>IWGSC在68个国家有2400个成员，是一个由小麦种植者、植物科学家、公共和私人育种家在2005年成立的一个国际合作联盟。IWGSC的愿景是获得面包小麦的高质量基因组序列，它是加速品种改良发展的基础。</p>
<p>为了实现此愿景，联盟采用来自捷克的Jaroslav Doležel教授高精湛的流式细胞仪分离技术，将普通小麦中国春（Chinese Spring）的21条染色体进行了分离。除了3B染色体之外，其余染色体均分为长臂（L）和短臂（S），并构建了相应的BAC文库，后续的物理图谱构建和BAC测序则由IWGSC的各成员国分担，我们国家西北农林科技大学主要是负责7DL物理图谱构建与BAC测序。</p>
<figure data-type="image" tabindex="1"><img src="https://tva1.sinaimg.cn/large/734cbfa0ly1g9ilhamliej20m70ku19p.jpg" alt="image" loading="lazy"></figure>
<h2 id="2-iwgsc的四个关键里程碑">2. IWGSC的四个关键里程碑</h2>
<p>面包小麦是世界上种植面积最大的小麦栽培作物，覆盖世界小麦种植区域的95%以上。获取它的基因组序列对于基因的改良是至关重要的。IWGSC采用分步走的策略，以确保小麦育种和小麦产业能够与小麦基因组基础研究并行并立即获得显著产出。IWGSC在进行小麦基因组组装过程中四个里程碑：</p>
<ol>
<li>利用21个面包小麦染色体的基因组survey序列,实现对小麦基因的每条染色体的定位；</li>
<li>开发物理地图，为测序提供资源；</li>
<li>建立每个染色体一个参考序列；</li>
<li>通过整合基于染色体的基因组资源与IWGSC组装的全基因组序列，获得金标准的小麦参考基因组序列。</li>
</ol>
<p>第一个里程碑是2014年7月发表在《科学》杂志上的以染色体为基础的基因组草图序列。所有染色体的物理图谱（里程碑2）在2015年底完成。在2016年1月底，全基因组鸟枪法获得每个染色体（IWGSC WGAV0.4-里程碑3）参考序列，这是在2016年6月前已经公布的。通过整合基于染色体图谱和序列资源与全基因组组装（IWGSC RefSeq v1.0）获得一个高质量的被注释的参考基因组序列，已在2017年1月公布，但限制使用。在2018年7月，所有IWGSC参考序列资源被公开提供，任何科研工作者均可以使用。</p>
<figure data-type="image" tabindex="2"><img src="https://tvax3.sinaimg.cn/large/734cbfa0ly1g9ilhjbfwsj20ky0cb49j.jpg" alt="image" loading="lazy"></figure>
<p>小麦相关数据主要位于URGI网站上：https://wheat-urgi.versailles.inra.fr/</p>
<figure data-type="image" tabindex="3"><img src="https://tva2.sinaimg.cn/large/734cbfa0ly1g9ilhql185j20qd0gxapw.jpg" alt="image" loading="lazy"></figure>
<h2 id="3-基因组测序材料-中国春品种">3. 基因组测序材料-中国春品种</h2>
<p>普通小麦是由祖先野生的一粒小麦(乌拉尔图小麦，含AA基因组)与拟斯卑尔托山羊草(Aegilops speltoides，含BB基因组)杂交形成四倍体野生二粒小麦(Triticum turgidum，含有AABB基因组)。在大约8000以前，亚洲西部肥沃新月地带，野生粗山羊草双二倍体(2n = 14; DD)和四倍体小麦 (2n = 4x = 28; AABB)自发杂交进而产生了六倍体小麦(T. aestivum; 2n = 6x = 42; AABBDD)。六倍体小麦的各祖先种均已完成测序，相关文献见下文。</p>
<figure data-type="image" tabindex="4"><img src="https://tva1.sinaimg.cn/large/734cbfa0ly1g9ilhxy0kgj20he0bk0y9.jpg" alt="image" loading="lazy"></figure>
<p>此次六倍体小麦测序品种选择了中国春，顾名思义，其来源肯定是中国呀。那么为什么全世界会选择它作为测序材料呢？</p>
<p>因为它有非常重要的遗传价值。中国春是种植在四川成都平原的一个地方品种，最早是通过传教士传入西方的，起初被称作为Chinese White，后来被改为Chinese Spring，即中国春。早在20世纪初，学者就发现中国春与黑麦非常容易杂交，并且利用中国春材料可以制造出单体、单端体、双端体、缺体、三体、缺体一四体等一系列非整倍体材料，推动了小麦遗传学研究领域的发展。后来研究者们又以人工的方式使中国春突变得到能够诱导部分同源染色体配对的基因突变体phlb，并被广泛地应用于远缘杂交进行小麦外源基因转移。因此中国春系列材料成了众所周知的小麦遗传学和外源基因转移等研究的工具和对照材料（摘自：刘登才, 郑有良, &amp; 兰秀锦. (2003). 小麦中国春遗传背景的育种改良. 中国农业科学, 36(11), 1383-1389.），而选择它作为基因组测序材料，也是必然的。</p>
<h2 id="4-文章解读">4. 文章解读</h2>
<p>研究背景</p>
<p>小麦是作为世界各地最为广泛种植的谷物之一，提供了19%人类能量消耗，相较于其他作物，小麦是唯一一个尚未完成全基因组测序的物种。究其原因，小麦基因组体量庞大，是人类基因组的5倍，是水稻基因组的40倍，作为典型的异源多倍体基因组，小麦基因组由三套相似而又不同的基因组整合在一起，最终形成一个极为复杂的六倍体基因组，同时其重复序列高，给研究工作带来巨大障碍。</p>
<p>研究结果<br>
<img src="https://tvax4.sinaimg.cn/large/734cbfa0ly1g9ili5sgizj20cn0ea10v.jpg" alt="image" loading="lazy"></p>
<ol>
<li>研究以小麦模式品系CS（Chinese Spring）“中国春”为材料，整合现有的各种数据资源，包括遗传图谱，以及单染色体BAC序列信息等。重新修正基因组大小，推测为为15.4~15.8Gb；利用DeNovoMAGIC2(来自NRgene公司)软件进行短片段组装，然后利用Hi-C技术和高密度遗传图谱（POPSEQ）进行染色体挂载。错误的组装的序列利用CSS（chromosome survey sequence）、Hi-C和POPSEQ进行识别和断开纠正。另外还测了大量的8-16fold的BAC，这些对于连接scaffold组成超大scaffold起了重要作用。 最终拼装获得了1,601条super-scaffold，大小为~14.5 GB， scaffold N50为22.8 Mb 的小麦基因组序列。研究中对A、B及D三套异源染色体组进行了严格区分，其中97%的序列被定位到ABD亚基因组上，只有大约481Mb的序列没有挂载到染色体。</li>
</ol>
<figure data-type="image" tabindex="5"><img src="https://tvax2.sinaimg.cn/large/734cbfa0ly1g9iliil8ouj20kj0k5qkf.jpg" alt="image" loading="lazy"></figure>
<p>文章对获得的基因组进行了较为严格的评估。</p>
<blockquote>
<p>1.遗传图评估，利用A、B、D的遗传图和染色体进行共线性评估，其中平均相关系数在0.98。</p>
</blockquote>
<blockquote>
<p>2.利用了7832个genetic position和4745个GBS数据得到的遗传标记跟染色体上位置的对应分析，其中相关系数在0.986和0.987。</p>
</blockquote>
<blockquote>
<p>3.1.24百万个neighbor insertion site（ISBPs）其中97%在BAC和基因组直接存在共线性（误差2Kb内）</p>
</blockquote>
<blockquote>
<p>4.从CSS或者ISBP中获的编码或者非编码的基因序列，在基因组中出现的比例分别为99%和97%。</p>
</blockquote>
<p>可以说这是迄今为止完成度最高且质量值最好的小麦基因组序列。</p>
<ol start="2">
<li>
<p>研究整合了21条小麦染色体参考序列、物理图谱及多种公共资源数据，获得了超过470万个分子标记，该数据将极大的推动后续的分子育种水平上的应用。</p>
</li>
<li>
<p>利用Hi-C染色质构象捕获技术及CSS数据，首次对小麦着丝粒区间大小进行了估算，大小估值为4.9MB。整个基因组重复序列含量为85%，三套亚基因组之间无明显差异。</p>
</li>
<li>
<p>在完整拼装的染色体序列基础上，注释并获得了共107,891个小麦基因，较此前公开的小麦基因组注释信息，这是最为完整和全面的小麦基因组注释版本。注释流程部分分别采用了TriAnnot和MIPS/PGSB两套流程，然后利用Mikado进行整合。其中得到了107,891个高可信度的基因，此外还有161,537个低可信度的基因。BUSCO评估有3个complete copy的占比达到了90%，至少一个complete copy的占到了99%，证明基因组与基因预测的完整性很高。</p>
</li>
</ol>
<figure data-type="image" tabindex="6"><img src="https://tva1.sinaimg.cn/large/734cbfa0ly1g9ilj0g70yj20f90lfqa0.jpg" alt="image" loading="lazy"></figure>
<p>此外通过探究了同源基因的亚基因组的分布模式及动态演化规律，结果发现三套染色体组之间的同源基因比例基本持平，鉴于D基因组进入小麦的时间较为落后，其同源基因含量水平相对较高，且假基因数量较少。同时小麦中未发现因植物多倍化过程中基因的缺失将导致基因组间的分歧显现，小麦基因组中的基因丢失及基因的位移是与其杂交事件发生的先后顺序的时间跨度成正比的。</p>
<figure data-type="image" tabindex="7"><img src="https://tva4.sinaimg.cn/large/734cbfa0ly1g9iljbyw0mj20ix088wl3.jpg" alt="image" loading="lazy"></figure>
<ol start="5">
<li>该研究整合了850个转录组数据，探究了小麦不同生长发育时期及生态环境下的表达谱模式，通过共表达网络模块分析及GO term富集分析，揭示了一系列开花时间相关的上下游基因。</li>
</ol>
<figure data-type="image" tabindex="8"><img src="https://tva3.sinaimg.cn/large/734cbfa0ly1g9iljhod3lj20gf0k5wsf.jpg" alt="image" loading="lazy"></figure>
<ol start="6">
<li>
<p>研究对小麦特有的扩张的基因家族进行了功能富集分析，结果发现扩张的基因显著富集在籽粒产量，生物非生物胁迫逆境胁迫相应，花粉育性等类组之下，这与人工育种中倾向与选择适应不同气候环境并具高产量的小麦品种的倾向相关联。</p>
</li>
<li>
<p>研究对响应非生物胁迫及抗虫性的SSt1 QTL基因进行剖析，以此为例探究了该小麦基因组在实际生产过程中的应用价值。</p>
</li>
</ol>
<p>小麦染色体基因组测序和注释的全面完成，可以大大加快研究小麦育种研究的步伐，有利于培育高产、抗旱、抗盐、抗病虫的小麦新品种，更好全球气候变化及人口膨胀带来的食品短缺的挑战，为小麦基因组改良工作奠定了基础。</p>
<h2 id="5-数据资源">5. 数据资源</h2>
<p>IWGSC RefSeq v1.0参考基因组、注释信息及其余该文章相关数据均可从https://wheat-urgi.versailles.inra.fr/Seq-Repository下载获得，测序原始数据存于SRA下登录号为SRP114784，全染色体/染色体臂BAC文库数据信息位于https://cnrgv.toulouse.inra.fr/en/Library/Wheat。</p>
<h2 id="6-小麦基因组主要已发表文献">6. 小麦基因组主要已发表文献</h2>
<p>小编总结目前二倍体、四倍体和六倍体小麦的组装情况</p>
<figure data-type="image" tabindex="9"><img src="https://tva4.sinaimg.cn/large/734cbfa0ly1g9iljq0o2dj20lj0770tn.jpg" alt="image" loading="lazy"></figure>
<ol>
<li>
<p>Brenchley, R., Spannagl, M., Pfeifer, M., Barker, G. L., D'Amore, R., &amp; Allen, A. M., et al. (2012). Analysis of the bread wheat genome using whole-genome shotgun sequencing. Nature, 491(7426), 705-710.</p>
</li>
<li>
<p>Jia, J., Zhao, S., Kong, X., Li, Y., Zhao, G., He, W., ... &amp; Jing, R. (2013). Aegilops tauschii draft genome sequence reveals a gene repertoire for wheat adaptation. Nature, 496(7443), 91.</p>
</li>
<li>
<p>Ling, H. Q., Zhao, S., Liu, D., Wang, J., Sun, H., Zhang, C., ... &amp; Gao, C. (2013). Draft genome of the wheat A-genome progenitor Triticum urartu. Nature, 496(7443), 87.</p>
</li>
<li>
<p>International Wheat Genome Sequencing Consortium. (2014). A chromosome-based draft sequence of the hexaploid bread wheat (Triticum aestivum) genome. Science, 345(6194), 1251788.</p>
</li>
<li>
<p>Choulet, F., Alberti, A., Theil, S., Glover, N., Barbe, V., &amp; Daron, J., et al. (2014). Structural and functional partitioning of bread wheat chromosome 3b. Science, 345(6194), 1249721.</p>
</li>
<li>
<p>Chapman, J. A., Mascher, M., Buluç, A., Barry, K., Georganas, E., Session, A., ... &amp; Schmutz, J. (2015). A whole-genome shotgun approach for assembling and anchoring the hexaploid bread wheat genome. Genome biology, 16(1), 26.</p>
</li>
<li>
<p>Zimin, A. V., Puiu, D., Hall, R., Kingan, S., Clavijo, B. J., &amp; Salzberg, S. L. (2017). The first near-complete assembly of the hexaploid bread wheat genome, Triticum aestivum. Gigascience.</p>
</li>
<li>
<p>Clavijo, B. J., Venturini, L., Schudoma, C., Accinelli, G. G., Kaithakottil, G., Wright, J., ... &amp; Lipscombe, J. (2017). An improved assembly and annotation of the allohexaploid wheat genome identifies complete families of agronomic genes and provides genomic evidence for chromosomal translocations. Genome research.</p>
</li>
<li>
<p>Zimin, A. V., Puiu, D., Luo, M. C., Zhu, T., Koren, S., Marçais, G., ... &amp; Salzberg, S. L. (2017). Hybrid assembly of the large and highly repetitive genome of Aegilops tauschii, a progenitor of bread wheat, with the MaSuRCA mega-reads algorithm. Genome research.</p>
</li>
<li>
<p>Avni, R., Nave, M., Barad, O., Baruch, K., Twardziok, S. O., Gundlach, H., ... &amp; Jordan, K. W. (2017). Wild emmer genome architecture and diversity elucidate wheat evolution and domestication. Science, 357(6346), 93-97.</p>
</li>
<li>
<p>Luo, M. C., Gu, Y. Q., Puiu, D., Wang, H., Twardziok, S. O., Deal, K. R., ... &amp; McGuire, P. E. (2017). Genome sequence of the progenitor of the wheat D genome Aegilops tauschii. Nature, 551(7681).</p>
</li>
<li>
<p>Zhao, G., Zou, C., Li, K., Wang, K., Li, T., Gao, L., ... &amp; Jiang, W. (2017). The Aegilops tauschii genome reveals multiple impacts of transposons. Nature plants, 3(12), 946.</p>
</li>
<li>
<p>Ling, H. Q., Ma, B., Shi, X., Liu, H., Dong, L., Sun, H., ... &amp; Yu, Y. (2018). Genome sequence of the progenitor of wheat A subgenome Triticum urartu. Nature, 557(7705), 424.</p>
</li>
</ol>

              </div>
              <div class="post-footer">
  <ul class="post-copyright">
    <li class="post-copyright-author">
      <strong>本文作者：</strong>
      生信学渣
    </li>
    <li class="post-copyright-link">
      <strong>本文链接：</strong>
      <a href="https://wangxf133.github.io/post/science-chong-bang-orli-shi-13-nian-liu-bei-ti-xiao-mai-ji-yin-zu-tu-pu-zhong-wan-cheng-xiang-xi-jie-du" title="Science重磅|历时13年六倍体小麦基因组图谱终完成（详细解读）">Science重磅|历时13年六倍体小麦基因组图谱终完成（详细解读）</a>
    </li>
    <li class="post-copyright-license">
      <strong>版权声明： </strong>
      本博客所有文章除特别声明外，转载请注明出处！
    </li>
  </ul>
  <div class="tags">
    
      <a href="https://wangxf133.github.io/tag/A3h6Jlq6p"># iwgsc</a>
    
      <a href="https://wangxf133.github.io/tag/We7V91z05"># 基因组</a>
    
      <a href="https://wangxf133.github.io/tag/jOhsegpai"># 小麦</a>
    
  </div>
  <div class="nav">
    <div class="nav-prev">
      
        <i class="fa fa-chevron-left"></i>
        <a class="nav-pc-next" href="https://wangxf133.github.io/post/circos-pei-zhi-wen-jian">Circos配置文件</a class="nav-pc-next">
        <a class="nav-mobile-prev" href="https://wangxf133.github.io/post/circos-pei-zhi-wen-jian">上一篇</a>
      
    </div>
    <div class="nav-next">
      
        <a class="nav-pc-next" href="https://wangxf133.github.io/post/shell-yu-yan-jiao-ben-gou-jian-5-cheng-xian-shu-ju-yu-jie-guo-shu-chu">shell语言脚本构建5-呈现数据与结果输出</a>
        <a class="nav-mobile-next" href="https://wangxf133.github.io/post/shell-yu-yan-jiao-ben-gou-jian-5-cheng-xian-shu-ju-yu-jie-guo-shu-chu">下一篇</a>
        <i class="fa fa-chevron-right"></i>
      
    </div>
  </div>
</div>
              
                
                  <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css">
<script src="https://unpkg.com/gitalk/dist/gitalk.min.js"></script>

<div id="gitalk-container"></div>

<script>
  var gitalk = new Gitalk({
    clientID: '5debd0dd15fc74854e71',
    clientSecret: '45d803fcebed7f8b06629f3f1626d55f24a79c4f',
    repo: 'wangxf133.github.io',
    owner: 'wangxf133',
    admin: ['wangxf133'],
    id: (location.pathname).substring(0, 49),      // Ensure uniqueness and length less than 50
    distractionFreeMode: false  // Facebook-like distraction free mode
  })
  gitalk.render('gitalk-container')
</script>
                
                
              
            </div>
          </div>
        </div>
      </div>
      <div class="footer-box">
  <footer class="footer">
    <div class="copyright">
      © 2019-2020 <i class="fa fa-heart"></i> HsxyHao
    </div>
    <div class="poweredby">
      <div class="power-left">Power By<a href="https://github.com/hsxyhao"> HsxyHao</a></div>
      <div>Copy<a href="https://github.com/iissnan/hexo-theme-next"> Hexo Next Theme</a></div>
    </div>
  </footer>
  <div class="back-to-top" id="back_to_top">
    <i class="fa fa-arrow-up"></i>
    <span class="scrollpercent">
      <span id="back_to_top_text">100</span>%
    </span>
  </div>
</div>

<script>

  let body = document.body;

  let back2Top = document.querySelector('#back_to_top'),
  back2TopText = document.querySelector('#back_to_top_text');

  function scrollAnimation(currentY, targetY) {
   
    let needScrollTop = targetY - currentY
    let _currentY = currentY
    setTimeout(() => {
      const dist = Math.ceil(needScrollTop / 10)
      _currentY += dist
      window.scrollTo(_currentY, currentY)
      if (needScrollTop > 10 || needScrollTop < -10) {
        scrollAnimation(_currentY, targetY)
      } else {
        window.scrollTo(_currentY, targetY)
      }
    }, 1)
  }

  back2Top.addEventListener("click", function(e) {
    scrollAnimation(body.scrollTop, 0);
    e.stopPropagation();
    return false;
  });
  
  window.addEventListener('scroll', function(e) {
    let percent = body.scrollTop / (body.scrollHeight - body.clientHeight) * 100;
    if (percent > 1 && !back2Top.classList.contains('back-top-active')) {
      back2Top.classList.add('back-top-active');
    }
    if (percent == 0) {
      back2Top.classList.remove('back-top-active');
    }
    back2TopText.textContent = Math.floor(percent);
  });

  // 动态调整站点概览高度
  let sideBar = document.querySelector('.sidebar');
  let navUl = document.querySelector('#nav_ul');
  sideBar.style = 'margin-top:' + (navUl.children.length * 42 + 162) + 'px; opacity: 1;';
  // 代码高亮
  hljs.initHighlightingOnLoad();
</script>
    </div>
  </body>
  <script src="/media/js/motion.js"></script>
</html>