<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" >

<title>R_DESeq2差异基因分析 | xiaofei</title>
<meta name="description" content="东北师范大学-在读博士">

<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">

<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.2/css/all.css" integrity="sha384-fnmOCqbTlWIlj8LyTjo7mOUStjsKC4pOpQbqyi7RrhN7udi9RwhKkMHpvLbHG9Sr" crossorigin="anonymous">
<link rel="shortcut icon" href="https://wangxf133.github.io/favicon.ico?v=1555248108171">
<link rel="stylesheet" href="https://wangxf133.github.io/styles/main.css">


  
    <link rel="stylesheet" href="https://unpkg.com/gitalk/dist/gitalk.css" />
  

  


<script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
<script src="https://cdn.bootcss.com/highlight.js/9.12.0/highlight.min.js"></script>

<link rel="stylesheet" href="https://unpkg.com/aos@next/dist/aos.css" />



  </head>
  <body>
    <div id="app" class="main">

      <div class="sidebar" :class="{ 'full-height': menuVisible }">
  <div class="top-container" data-aos="fade-right">
    <div class="top-header-container">
      <a class="site-title-container" href="https://wangxf133.github.io">
        <img src="https://wangxf133.github.io/images/avatar.png?v=1555248108171" class="site-logo">
        <h1 class="site-title">xiaofei</h1>
      </a>
      <div class="menu-btn" @click="menuVisible = !menuVisible">
        <div class="line"></div>
      </div>
    </div>
    <div>
      
        
          <a href="/" class="site-nav">
            首页
          </a>
        
      
        
          <a href="/archives" class="site-nav">
            归档
          </a>
        
      
        
          <a href="/tags" class="site-nav">
            标签
          </a>
        
      
        
          <a href="/post/about" class="site-nav">
            关于
          </a>
        
      
    </div>
  </div>
  <div class="bottom-container" data-aos="flip-up" data-aos-offset="0">
    <div class="social-container">
      
        
          <a class="social-link" href="wangxf133@nenu.edu.cn" target="_blank">
            <i class="fab fa-github"></i>
          </a>
        
      
        
      
        
      
        
          <a class="social-link" href="https://www.zhihu.com/people/wang-xiao-fei-66-3/activities" target="_blank">
            <i class="fab fa-zhihu"></i>
          </a>
        
      
        
      
    </div>
    <div class="site-description">
      东北师范大学-在读博士
    </div>
    <div class="site-footer">
      Powered by Gridea
    </div>
  </div>
</div>


      <div class="main-container">
        <div class="content-container" data-aos="fade-up">
          <div class="post-detail">
            <h2 class="post-title">R_DESeq2差异基因分析</h2>
            <div class="post-date">2019-03-27</div>
            
            <div class="post-content">
              <!--more-->
<p>DESeq2和EdgeR可用于做基因差异表达分析，主要也是用于RNA-Seq数据，同样也可以处理类似的ChIP-Seq,shRNA以及质谱数据。</p>
<p>这两个都属于R包，其相同点在于都是对count data数据进行处理，都是基于负二项分布模型。因此会发现，用两者处理同一组数据，最后在相同阈值下筛选出的大部分基因都是一样的，但是有一部分不同应该是由于其估计离散度的不同方法所导致的。</p>
<p>不同样品的测序量会有差异，最简单的标准化方式是计算
counts per million (CPM)，即原始reads count除以总reads数乘以1,000,000。</p>
<p>这种计算方式的缺点是容易受到极高表达且在不同样品中存在差异表达的基因的影响；这些基因的打开或关闭会影响到细胞中总的分子数目，可能导致这些基因标准化之后就不存在表达差异了，而原本没有差异的基因标准化之后却有差异了。</p>
<p>RPKM、FPKM和TPM是CPM按照基因或转录本长度归一化后的表达，也会受到这一影响。</p>
<pre><code>calc_cpm &lt;- function (expr_mat, spikes = NULL){
  norm_factor &lt;- colSums(expr_mat[-spikes,])
  return(t(t(expr_mat)/norm_factor)) * 10^6
}
</code></pre>
<p>为了解决这一问题，研究者提出了其它的标准化方法。</p>
<p><code>量化因子 (size factor, SF)</code>是由DESeq提出的。其方法是首先计算每个基因在所有样品中表达的几何平均值。每个细胞的量化因子(size
factor)是所有基因与其在所有样品中的表达值的几何平均值的比值的中位数。由于几何平均值的使用，只有在所有样品中表达都不为0的基因才能用来计算。这一方法又被称为 <code>RLE (relative log expression)</code>。</p>
<pre><code>calc_sf &lt;- function (expr_mat, spikes=NULL){
  geomeans &lt;- exp(rowMeans(log(expr_mat[-spikes,])))
  SF &lt;- function(cnts){
    median((cnts/geomeans)[(is.finite(geomeans) &amp; geomeans &gt;0)])
  }
  norm_factor &lt;- apply(expr_mat[-spikes,],2,SF)
  return(t(t(expr_mat)/norm_factor))
}
</code></pre>
<p><code>上四分位数 (upperquartile, UQ)</code>是样品中所有基因的表达除以处于上四分位数的基因的表达值。同时为了保证表达水平的相对稳定，计算得到的上四分位数值要除以所有样品中上四分位数值的中位数。</p>
<pre><code>calc_uq &lt;- function (expr_mat, spikes=NULL){
  UQ &lt;- function(x) {
    quantile(x[x&gt;0],0.75)
  }
  uq &lt;- unlist(apply(expr_mat[-spikes,],2,UQ))
  norm_factor &lt;- uq/median(uq)
  return(t(t(expr_mat)/norm_factor))
}
</code></pre>
<p>TMM是M-值的加权截尾均值。选定一个样品为参照，其它样品中基因的表达相对于参照样品中对应基因表达倍数的log2值定义为M-值。随后去除M-值中最高和最低的30%，剩下的M值计算加权平均值。每一个非参照样品的基因表达值都乘以计算出的TMM。这个方法假设大部分基因的表达是没有差异的。</p>
<h1 id="deseq2-工作流程">DESeq2 工作流程</h1>
<h2 id="1安装包-deseq2安装方法如下">1.安装包 DESeq2安装方法如下</h2>
<pre><code>source(&quot;https://bioconductor.org/biocLite.R&quot;)
biocLite('BiocInstaller')
biocLite(&quot;DESeq2&quot;)
install.packages(c(&quot;gplots&quot;, &quot;amap&quot;, &quot;ggplot2&quot;))
</code></pre>
<h2 id="2加载包">2.加载包</h2>
<pre><code>library(DESeq2)

## Loading required package: S4Vectors

## Loading required package: stats4

## Loading required package: BiocGenerics

## Loading required package: parallel

##
## Attaching package: 'BiocGenerics'

## The following objects are masked from 'package:parallel':
##
##     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
##     clusterExport, clusterMap, parApply, parCapply, parLapply,
##     parLapplyLB, parRapply, parSapply, parSapplyLB

## The following objects are masked from 'package:stats':
##
##     IQR, mad, sd, var, xtabs

## The following objects are masked from 'package:base':
##
##     anyDuplicated, append, as.data.frame, cbind, colMeans,
##     colnames, colSums, do.call, duplicated, eval, evalq, Filter,
##     Find, get, grep, grepl, intersect, is.unsorted, lapply,
##     lengths, Map, mapply, match, mget, order, paste, pmax,
##     pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce,
##     rowMeans, rownames, rowSums, sapply, setdiff, sort, table,
##     tapply, union, unique, unsplit, which, which.max, which.min

##
## Attaching package: 'S4Vectors'

## The following object is masked from 'package:base':
##
##     expand.grid

## Loading required package: IRanges

## Loading required package: GenomicRanges

## Loading required package: GenomeInfoDb

## Loading required package: SummarizedExperiment

## Loading required package: Biobase

## Welcome to Bioconductor
##
##     Vignettes contain introductory material; view with
##     'browseVignettes()'. To cite Bioconductor, see
##     'citation(&quot;Biobase&quot;)', and for packages 'citation(&quot;pkgname&quot;)'.

## Loading required package: DelayedArray

## Loading required package: matrixStats

##
## Attaching package: 'matrixStats'

## The following objects are masked from 'package:Biobase':
##
##     anyMissing, rowMedians

##
## Attaching package: 'DelayedArray'

## The following objects are masked from 'package:matrixStats':
##
##     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges

## The following object is masked from 'package:base':
##
##     apply

library(&quot;RColorBrewer&quot;)
library(&quot;gplots&quot;)

##
## Attaching package: 'gplots'

## The following object is masked from 'package:IRanges':
##
##     space

## The following object is masked from 'package:S4Vectors':
##
##     space

## The following object is masked from 'package:stats':
##
##     lowess

library(&quot;amap&quot;)
library(&quot;ggplot2&quot;)
library(&quot;BiocParallel&quot;)

# 多线程加速, 使用4个核
# 如果你电脑性能强大，可以把这个数变大
register(MulticoreParam(4))
</code></pre>
<h2 id="3读入数据">3.读入数据</h2>
<pre><code># 注意文件的位置，程序自己不知道文件在什么地方
# 如果只给文件名，不给路径，程序只会在当前目录查找文件
# 若找不到，则会报错
# 可以使用下面命令设置工作目录到文件所在目录
# 或提供文件全路径
# setwd(&quot;D:/R_lianxi/DEseq2&quot;)
data &lt;- read.table(&quot;A4vsWT.txt&quot;,header = T , row.names = 1 ,sep = &quot;\t&quot;)


# 预过滤
# 撇掉在多于两个样本中count&lt;1的值，如果样本数多，这个数值可以适当增加
# 排除极低表达基因的干扰
data &lt;- data[rowSums(data)&gt;2,]
head(data)

##                 untrt_N61311 untrt_N052611 untrt_N080611 untrt_N061011
## ENSG00000227232           13            25            23            24
## ENSG00000278267            0             5             3             4
## ENSG00000268903            0             2             0             0
## ENSG00000269981            0             3             0             1
## ENSG00000241860            3            11             1             5
## ENSG00000279457           46            90            73            49
##                 trt_N61311 trt_N052611 trt_N080611 trt_N061011
## ENSG00000227232         12          12          22          22
## ENSG00000278267          2           4           3           1
## ENSG00000268903          0           2           0           3
## ENSG00000269981          0           2           0           0
## ENSG00000241860          3           2           0           2
## ENSG00000279457         52          46          89          31
</code></pre>
<pre><code># 分组信息1---这个是读入一个文件
sample &lt;- read.table(&quot;sampleFile_A4WT.txt&quot;, header=T, row.names=1, sep=&quot;\t&quot;)

sample &lt;- sample[match(colnames(data), rownames(sample)),, drop=F]
## Match参数 default: match(x, table, nomatch = NA_integer_,incomparables = NULL)，match（x, y）返回的是vector x中每个元素在vectory中对映的位置（positions in y），如果vector x中存在不在vectory中的元素，该元素处返回的是NA，incomparables选项在x，y类型不同时使用，现在对了类型的强制转换并不熟悉，并且这个很少会用到，一般是同类型的比较。
## 上面这个主要是看counts数据的行名和分组信息中的列名是否一致
sample_rowname &lt;- rownames(sample)
sample_rowname


# 分组信息2---这个是直接在代码中设定
data &lt;- data
conditions &lt;- factor(c(&quot;tre&quot;,&quot;tre&quot;,&quot;con&quot;,&quot;con&quot;))
sample &lt;- data.frame(row.names=colnames(data), conditions)
</code></pre>
<pre><code># 下面的可以忽略，如果没遇到错误不需要执行
# 目的是做因子转换
sample &lt;- data.frame(lapply(sample, function(x) factor(x, levels=unique(x))))
rownames(sample) &lt;- sample_rowname
sample

#     conditions
#A4_1        tre
#A4_3        tre
#WT_1        con
#WT_3        con
</code></pre>
<h2 id="4产生deseq数据集">4.产生DESeq数据集</h2>
<p>DESeq2包采用<code>DESeqDataSet</code>存储原始的<code>read count</code>和中间计算的统计量。</p>
<p>每个<code>DESeqDataSet</code>对象都要有一个实验设计formula，用于对数据进行分组，以便计算表达值的离散度和估计表达倍数差异，通常格式为<code>~ batch + conditions</code>
(为了方便后续计算，最为关注的分组信息放在最后一位)。
<code>~在R里面用于构建公式对象，~左边为因变量，右边为自变量</code></p>
<p>countData: 表达矩阵</p>
<ul>
<li>即下面代码中的countData，就是我们前面通过read count计算后并融合生成的矩阵，行为各个基因，列为各个样品，中间为计算reads或者fragment得到的整数。我们后面要用的是这个文件.</li>
</ul>
<p>colData: 样品分组信息表</p>
<ul>
<li>样品信息矩阵即下面代码中的colData，它的类型是一个dataframe（数据框），第一列是样品名称，第二列是样品的处理情况（对照还是处理等），即condition，condition的类型是一个factor。这些信息可以从导出，也可以自己单独建立。</li>
</ul>
<p>design: 实验设计信息，conditions必须是colData中的一列</p>
<ul>
<li>实验设计信息，conditions必须是colData中的一列</li>
<li>上述代码中的design。 差异比较矩阵就是告诉差异分析函数是要从要分析哪些变量间的差异，简单说就是说明哪些是对照哪些是处理。</li>
</ul>
<pre><code>ddsFullCountTable &lt;- DESeqDataSetFromMatrix(countData = data,
        colData = sample,  design= ~ conditions)


dds &lt;- DESeq(ddsFullCountTable)
#使用DESeq函数进行估计离散度，然后进行标准的差异表达分析，得到res对象结果
#相关说明： DESeq包含三步，estimation of size factors（estimateSizeFactors)， estimation of dispersion（estimateDispersons)， Negative Binomial GLM fitting and Wald statistics（nbinomWaldTest），可以分布运行，也可用一步到位，最后返回 results可用的DESeqDataSet对象。

## estimating size factors

## estimating dispersions

## gene-wise dispersion estimates

## mean-dispersion relationship

## final dispersion estimates

## fitting model and testing
</code></pre>
<h2 id="5获取标准化后的数据">5.获取标准化后的数据</h2>
<pre><code># ?counts查看此函数功能
# normalized=T, 返回标准化的数据
normalized_counts &lt;- counts(dds, normalized=TRUE)
head(normalized_counts)
#                 A4_1       A4_3      WT_1       WT_3
# AT1G01010   48.78565   60.06335  70.30580   82.48226
# AT1G01020  176.84798  222.64862 182.02462  173.69794
# AT1G01040 1042.79325 1032.46826 947.68371 1011.13549
# AT1G01050  729.75200  645.16322 708.83660  693.82138
# AT1G01060  294.74663  269.24950 302.41127  336.72170
</code></pre>
<p>根据基因在不同的样本中表达变化的差异程度mad值对数据排序，差异越大的基因排位越前。</p>
<pre><code>normalized_counts_mad &lt;- apply(normalized_counts, 1, mad)
normalized_counts &lt;- normalized_counts[order(normalized_counts_mad, decreasing=T), ]


# 标准化后的数据输出
write.table(normalized_counts, file=&quot;ehbio_trans.Count_matrix.xls.DESeq2.normalized.xls&quot;,
quote=F, sep=&quot;\t&quot;, row.names=T, col.names=T)

# 标准化后的数据输出
write.table(normalized_counts, file=&quot;ehbio_trans.Count_matrix.xls.DESeq2.normalized.xls&quot;,
quote=F, sep=&quot;\t&quot;, row.names=T, col.names=T)

# 只在Linux下能运行，目的是填补表格左上角内容
system(paste(&quot;sed -i '1 s/^/ID\t/'&quot;, &quot;ehbio_trans.Count_matrix.xls.DESeq2.normalized.xls&quot;))

# log转换后的结果
rld &lt;- rlog(dds, blind=FALSE)
rlogMat &lt;- assay(rld)
rlogMat &lt;- rlogMat[order(normalized_counts_mad, decreasing=T), ]

write.table(rlogMat, file=&quot;ehbio_trans.Count_matrix.xls.DESeq2.normalized.rlog.xls&quot;,
quote=F, sep=&quot;\t&quot;, row.names=T, col.names=T)

# 只在Linux下能运行，目的是填补表格左上角内容
system(paste(&quot;sed -i '1 s/^/ID\t/'&quot;, &quot;ehbio_trans.Count_matrix.xls.DESeq2.normalized.rlog.xls&quot;))
</code></pre>
<p>样品层级聚类分析，判断样品的相似性和组间组内差异</p>
<pre><code># 生成颜色
hmcol &lt;- colorRampPalette(brewer.pal(9, &quot;GnBu&quot;))(100)

# 计算相关性pearson correlation
pearson_cor &lt;- as.matrix(cor(rlogMat, method=&quot;pearson&quot;))

# 层级聚类
hc &lt;- hcluster(t(rlogMat), method=&quot;pearson&quot;)

# 热图绘制
## 在命令行下运行时，需要去除下面开头的#号，把输出的图保存到文件中
## 输出到文件时，dev.off()命令是关闭输出，从而完成图片的写入。如果不做这一步，则图片则不能写入，从而不能打开
## 如果在Rstudio或其它可视化界面时，可以直接把图输出到屏幕
#pdf(&quot;ehbio_trans.Count_matrix.xls.DESeq2.normalized.rlog.pearson.pdf&quot;, pointsize=10)
heatmap.2(pearson_cor, Rowv=as.dendrogram(hc), symm=T, trace=&quot;none&quot;,
col=hmcol, margins=c(11,11), main=&quot;The pearson correlation of each
sample&quot;)
</code></pre>
<p><img src="https://ws1.sinaimg.cn/large/734cbfa0ly1frwyg6p65oj20nl0h4dle.jpg" alt=""></p>
<pre><code>#dev.off()
</code></pre>
<p>样品PCA分析</p>
<pre><code>pca_data &lt;- plotPCA(rld, intgroup=c(&quot;conditions&quot;), returnData=T, ntop=5000)
</code></pre>
<h2 id="6差异基因分析">6.差异基因分析</h2>
<pre><code># 定义变量的好处是下面的代码就独立了
# 如果想比较不同的组，只需在这修改即可，后面代码都可以不动
sampleA = &quot;tre&quot;
sampleB = &quot;con&quot;
</code></pre>
<p><code>results函数</code>提取差异基因分析结果，包含<code>log2 fold changes</code>,
<code>p values</code>和<code>adjusted p values</code>.</p>
<p><code>constrast</code>用于指定比较的两组的信息，输出的log2FoldChange为<code>log2(SampleA/SampleB)</code> <strong>(重要啊！！！)</strong>。</p>
<pre><code>contrastV &lt;- c(&quot;conditions&quot;, sampleA, sampleB)
res &lt;- results(dds,  contrast=contrastV)
res
log2 fold change (MLE): conditions tre vs con
Wald test p-value: conditions tre vs con
DataFrame with 22852 rows and 6 columns
                  baseMean      log2FoldChange             lfcSE               stat            pvalue              padj
                 &lt;numeric&gt;           &lt;numeric&gt;         &lt;numeric&gt;          &lt;numeric&gt;         &lt;numeric&gt;         &lt;numeric&gt;
## AT1G01010 65.4092660581971  -0.489358795511224 0.521216004216664 -0.938879066552613 0.347792838086431 0.999976236609269
## AT1G01020 188.804788979982   0.167293274077293 0.362795264673978    0.4611230916358 0.644710295499361 0.999976236609269
## AT1G01030 35.9145116528229   0.127302421234151 0.672062276718896  0.189420572533336 0.849763201781784 0.999976236609269
## AT1G01040 1008.52017973907  0.0833172912410816 0.240007179688198  0.347144995201069 0.728482391586173 0.999976236609269
## AT1G01050 694.393299169352 -0.0287852240886701 0.259768675170891 -0.110810990084673 0.911766230200056 0.999976236609269
## ...                    ...                 ...               ...                ...               ...               ...
## AT5G67600 578.828536547001 -0.0573356586517927 0.289518404967545  -0.19803804410369  0.84301529826065 0.999976236609269
## AT5G67610 265.008387291835   0.107651655525928 0.325292335215263  0.330938186584351 0.740691176896291 0.999976236609269
## AT5G67620 10.3580543313242  -0.914229542297002  1.14234019766374 -0.800312852656979 0.423529558307148 0.999976236609269
## AT5G67630 300.480587343979  -0.362595118844209 0.397418243520207 -0.912376632820008  0.36157048603897 0.999976236609269
## AT5G67640  63.749399447105    0.46227365476905 0.536557711991677  0.861554394685917 0.388932776218197 0.999976236609269
</code></pre>
<p>给<code>DESeq2</code>的原始输出结果增加样品平均表达信息，使得结果更容易理解和解析。</p>
<pre><code># 获得第一组数据均值
baseA &lt;- counts(dds, normalized=TRUE)[, colData(dds)$conditions == sampleA]

if (is.vector(baseA)){
    baseMeanA &lt;- as.data.frame(baseA)
} else {
    baseMeanA &lt;- as.data.frame(rowMeans(baseA))
}
colnames(baseMeanA) &lt;- sampleA
head(baseMeanA)

#                 tre
# AT1G01010   54.42450
# AT1G01020  199.74830
# AT1G01030   37.50077
# AT1G01040 1037.63076
# AT1G01050  687.45761
# AT1G01060  281.99806
</code></pre>
<pre><code># 获得第二组数据均值
baseB &lt;- counts(dds, normalized=TRUE)[, colData(dds)$conditions == sampleB]
if (is.vector(baseB)){
        baseMeanB &lt;- as.data.frame(baseB)
} else {
        baseMeanB &lt;- as.data.frame(rowMeans(baseB))
}
colnames(baseMeanB) &lt;- sampleB
head(baseMeanB)

##                  con
##  AT1G01010  76.39403
##  AT1G01020 177.86128
##  AT1G01030  34.32825
##  AT1G01040 979.40960
##  AT1G01050 701.32899
##  AT1G01060 319.56649
</code></pre>
<pre><code># 结果组合
res &lt;- cbind(baseMeanA, baseMeanB, as.data.frame(res))
head(res)
##
##                tre       con   baseMean      log2FoldChange  lfcSE      stat    pvalue      padj
##  AT1G01010   54.42450  76.39403   65.40927    -0.48935880 0.5212160 -0.9388791 0.3477928 0.9999762
##  AT1G01020  199.74830 177.86128  188.80479     0.16729327 0.3627953  0.4611231 0.6447103 0.9999762
##  AT1G01030   37.50077  34.32825   35.91451     0.12730242 0.6720623  0.1894206 0.8497632 0.9999762
##  AT1G01040 1037.63076 979.40960 1008.52018     0.08331729 0.2400072  0.3471450 0.7284824 0.9999762
##  AT1G01050  687.45761 701.32899  694.39330    -0.02878522 0.2597687 -0.1108110 0.9117662 0.9999762
##  AT1G01060  281.99806 319.56649  300.78227    -0.18036321 0.3077631 -0.5860456 0.5578448 0.9999762
</code></pre>
<pre><code># 增加ID信息
res &lt;- cbind(ID=rownames(res), as.data.frame(res))
res$baseMean &lt;- rowMeans(cbind(baseA, baseB))

# 校正后p-value为NA的复制为1
res$padj[is.na(res$padj)] &lt;- 1

# 按pvalue排序, 把差异大的基因放前面
res &lt;- res[order(res$pvalue),]
head(res)
##                    ID        tre         con  baseMean log2FoldChange     lfcSE      stat       pvalue         padj
##   AT1G57750 AT1G57750   28.16139 1003.180258  515.6708      -5.153560 0.3642816 -14.14719 1.943736e-45 3.752770e-41
##   AT1G33760 AT1G33760  639.33137    6.767155  323.0493       6.561945 0.5524517  11.87786 1.542626e-32 1.489174e-28
##   AT1G76650 AT1G76650 4693.39043  165.804397 2429.5974       4.823094 0.4180847  11.53617 8.670392e-31 5.579975e-27
##   AT1G56660 AT1G56660 6742.93138  311.088739 3527.0101       4.437947 0.3916322  11.33193 9.117706e-30 4.400889e-26
##   AT1G15010 AT1G15010  488.46638   13.548882  251.0076       5.172803 0.4989248  10.36790 3.470615e-25 1.272240e-21
##   AT5G41750 AT5G41750 2075.48402  350.421899 1212.9530       2.566263 0.2478179  10.35544 3.953716e-25 1.272240e-21
</code></pre>
<p>整体分析结果输出到文件</p>
<pre><code># 生成文件名

comp314 &lt;- paste(sampleA, &quot;_vs_&quot;, sampleB, sep=&quot;.&quot;)
file_base &lt;- paste(&quot;ehbio_trans.Count_matrix.xls.DESeq2&quot;, comp314, sep=&quot;.&quot;)
file_base1 &lt;- paste(file_base, &quot;results.xls&quot;, sep=&quot;.&quot;)
write.table(as.data.frame(res), file=file_base1, sep=&quot;\t&quot;, quote=F, row.names=F)
</code></pre>
<p>提取差异表达基因</p>
<pre><code># 差异基因筛选，padj &lt; 0.1
res_DEG &lt;- subset(res, res$padj&lt;0.1, select=c('ID', sampleA,
        sampleB, 'log2FoldChange', 'padj'))
# foldchang &gt; 1
res_DEG_up &lt;- subset(res_DEG, res_DEG$log2FoldChange&gt;=1)

file &lt;- paste(&quot;ehbio_trans.Count_matrix.xls.DESeq2&quot;,sampleA,&quot;_higherThan_&quot;,sampleB,
        'xls', sep=&quot;.&quot;)
write.table(as.data.frame(res_DEG_up), file=file, sep=&quot;\t&quot;, quote=F, row.names=F)

res_DEG_down &lt;- subset(res_DEG, res_DEG$log2FoldChange&lt;=(-1)*1)

file &lt;- paste(&quot;ehbio_trans.Count_matrix.xls.DESeq2&quot;,sampleA, &quot;_lowerThan_&quot;, sampleB,
        'xls', sep=&quot;.&quot;)
write.table(as.data.frame(res_DEG_down), file=file, sep=&quot;\t&quot;, quote=F, row.names=F)

# 差异基因ID
res_de_up_id = data.frame(ID=res_DEG_up$ID,
        type=paste(sampleA,&quot;_higherThan_&quot;, sampleB, sep=&quot;.&quot;))
res_de_dw_id = data.frame(ID=res_DEG_down$ID,
        type=paste(sampleA,&quot;_lowerThan_&quot;, sampleB, sep=&quot;.&quot;))
DEG_id = rbind(res_de_up_id, res_de_dw_id)

file &lt;- &quot;ehbio_trans.Count_matrix.xls.DESeq2.all.DE&quot;
write.table(as.data.frame(DEG_id), file=file, sep=&quot;\t&quot;, quote=F, row.names=F, col.names=F)
</code></pre>
<p>绘制火山图</p>
<pre><code># 可以把前面生成的results.xls文件提交到www.ehbio.com/ImageGP绘制火山图
# 或者使用我们的s-plot https://github.com/Tong-Chen/s-plot
logCounts &lt;- log2(res$baseMean+1)
logFC &lt;- res$log2FoldChange
FDR &lt;- res$padj
#png(filename=paste(file_base, &quot;Volcano.png&quot;, sep=&quot;.&quot;))
plot(logFC, -1*log10(FDR), col=ifelse(FDR&lt;=0.01, &quot;red&quot;, &quot;black&quot;),
 xlab=&quot;logFC&quot;, ylab=&quot;-1*log1o(FDR)&quot;, main=&quot;Volcano plot&quot;, pch=&quot;.&quot;)
#dev.off()
</code></pre>
<p><img src="https://ws1.sinaimg.cn/large/734cbfa0ly1frwygx98ufj20nb0goq5d.jpg" alt=""></p>
<p>差异基因热图</p>
<pre><code>res_de_up_top20_id &lt;- as.vector(head(res_de_up$ID,20))
res_de_dw_top20_id &lt;- as.vector(head(res_de_dw$ID,20))

red_de_top20 &lt;- c(res_de_up_top20_id, res_de_dw_top20_id)
red_de_top20

##  [1] &quot;ENSG00000178695&quot; &quot;ENSG00000196517&quot; &quot;ENSG00000116584&quot;
##  [4] &quot;ENSG00000144369&quot; &quot;ENSG00000276600&quot; &quot;ENSG00000108821&quot;
##  [7] &quot;ENSG00000162692&quot; &quot;ENSG00000181467&quot; &quot;ENSG00000145777&quot;
## [10] &quot;ENSG00000163491&quot; &quot;ENSG00000183160&quot; &quot;ENSG00000172986&quot;
## [13] &quot;ENSG00000164484&quot; &quot;ENSG00000131389&quot; &quot;ENSG00000134253&quot;
## [16] &quot;ENSG00000124766&quot; &quot;ENSG00000227051&quot; &quot;ENSG00000146006&quot;
## [19] &quot;ENSG00000112837&quot; &quot;ENSG00000146592&quot; &quot;ENSG00000152583&quot;
## [22] &quot;ENSG00000148175&quot; &quot;ENSG00000179094&quot; &quot;ENSG00000134686&quot;
## [25] &quot;ENSG00000125148&quot; &quot;ENSG00000120129&quot; &quot;ENSG00000189221&quot;
## [28] &quot;ENSG00000109906&quot; &quot;ENSG00000101347&quot; &quot;ENSG00000162614&quot;
## [31] &quot;ENSG00000096060&quot; &quot;ENSG00000162616&quot; &quot;ENSG00000166741&quot;
## [34] &quot;ENSG00000183044&quot; &quot;ENSG00000164125&quot; &quot;ENSG00000198624&quot;
## [37] &quot;ENSG00000256235&quot; &quot;ENSG00000077684&quot; &quot;ENSG00000135821&quot;
## [40] &quot;ENSG00000164647&quot;

# 可以把矩阵存储，提交到www.ehbio.com/ImageGP绘制火山图
# 或者使用我们的s-plot https://github.com/Tong-Chen/s-plot
red_de_top20_expr &lt;- normalized_counts[rownames(normalized_counts) %in% red_de_top20,]
library(pheatmap)
pheatmap(red_de_top20_expr, cluster_row=T, scale=&quot;row&quot;, annotation_col=sample)
</code></pre>
<p><img src="https://ws1.sinaimg.cn/large/734cbfa0ly1frwyiae8z7j20mb0pctoe.jpg" alt=""></p>
<h2 id="7关于批次效应">7.关于批次效应</h2>
<p>见论坛讨论 http://www.ehbio.com/Esx/d/10-deseq2。</p>
<p>每个DESeqDataSet对象都要有一个实验设计formula，用于对数据进行分组，以便计算表达值的离散度和估计表达倍数差异，通常格式为~ batch + conditions
(为了方便后续计算，最为关注的分组信息放在最后一位)。</p>
<p>如果记录了样本的批次信息，或者其它需要抹除的信息可以定义在design参数中，在下游回归的分析中会根据design formula来估计batch effect的影响，并在下游分析时减去这个影响。这是处理batch effect的推荐方式。</p>
<p>在模型中考虑batch effect并没有在数据矩阵中移除bacth effect，如果下游处理时，确实有需要可以使用limma包的removeBatchEffect来处理。</p>
<p>countData: 表达矩阵</p>
<p>colData: 样品分组信息表</p>
<p>design: 实验设计信息，batch和conditions必须是colData中的一列</p>
<pre><code>ddsFullCountTable &lt;- DESeqDataSetFromMatrix(countData = data,
        colData = sample,  design= ~ batch + conditions)

dds &lt;- DESeq(ddsFullCountTable)

rld &lt;- rlog(dds, blind=FALSE)
rlogMat &lt;- assay(rld)
rlogMat &lt;- limma::removeBatchEffect(rlogMat, c(sample$batch))
</code></pre>
<p>Just to be clear, there’s an important difference between removing a batch effect and modelling a batch effect. Including the batch in your design formula will model the batch effect in the regression step, which means that the raw data are not modified (so the batch effect is not removed), but instead the regression will estimate the size of the batch effect and subtract it out when performing all other tests. In addition, the model’s residual degrees of freedom will be reduced appropriately to reflect the fact that some degrees of freedom were “spent” modelling the batch effects. This is the preferred approach for any method that is capable of using it (this includes DESeq2). You would only remove the batch effect (e.g. using limma’s removeBatchEffect function) if you were going to do some kind of downstream analysis that can’t model the batch effects, such as training a classifier.</p>
<p>批次效应模拟</p>
<pre><code>#Make some simulated data with a batch effect:
dds &lt;- makeExampleDESeqDataSet(betaSD=1,interceptMean=10)
dds$batch &lt;- factor(rep(c(&quot;A&quot;,&quot;B&quot;),each=6))

#VST, remove batch effect, then plotPCA:

vsd &lt;- vst(dds)
plotPCA(vsd, &quot;batch&quot;)
</code></pre>
<p>参考：</p>
<ul>
<li><a href="https://mp.weixin.qq.com/s?__biz=MzI5MTcwNjA4NQ==&amp;mid=2247485368&amp;idx=1&amp;sn=12b20487e9014ce2e69f01d3efbc6ce8&amp;chksm=ec0dc232db7a4b248530e117a86053a9149d9ede5258326e04a4ab375f6858853c2e4c06d07d&amp;mpshare=1&amp;scene=23&amp;srcid=0530Jk9L86jUiDQixEk1Znbh#rd">生信宝典</a></li>
<li>https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html</li>
<li>http://dx.doi.org/10.1186/s13059-014-0550-8</li>
<li>https://zhuanlan.zhihu.com/p/31952724</li>
<li>https://zhuanlan.zhihu.com/p/32668252</li>
<li>https://zhuanlan.zhihu.com/p/34812206</li>
<li>https://zhuanlan.zhihu.com/p/30350531</li>
<li>http://www.bioinfo-scrounger.com/archives/113</li>
</ul>

            </div>
            
              <div class="tag-container">
                
                  <a href="https://wangxf133.github.io/tag/AlmdaiUZE" class="tag">
                    DESeq2
                  </a>
                
                  <a href="https://wangxf133.github.io/tag/VWu5Tyyx-" class="tag">
                    RNA-seq
                  </a>
                
                  <a href="https://wangxf133.github.io/tag/6iQrV6R7j" class="tag">
                    R
                  </a>
                
                  <a href="https://wangxf133.github.io/tag/ANs9W6M5C" class="tag">
                    Bioconductor
                  </a>
                
              </div>
            
            
              <div class="next-post">
                <div class="next">下一篇</div>
                <a href="https://wangxf133.github.io/post/bioconductor-de-zhi-pu-dan-bai-zu-xue-shu-ju-fen-xi">
                  <h3 class="post-title">
                    Bioconductor的质谱蛋白组学数据分析
                  </h3>
                </a>
              </div>
            

            
              
                <div id="gitalk-container" data-aos="fade-in"></div>
              

              
            

          </div>

        </div>
      </div>
    </div>

    <script src="https://unpkg.com/aos@next/dist/aos.js"></script>

<script type="application/javascript">

AOS.init();

hljs.initHighlightingOnLoad()

var app = new Vue({
  el: '#app',
  data: {
    menuVisible: false,
  },
})

</script>


  
  
    <script src="https://unpkg.com/gitalk/dist/gitalk.min.js"></script>
    <script>

      var gitalk = new Gitalk({
        clientID: '5debd0dd15fc74854e71',
        clientSecret: '45d803fcebed7f8b06629f3f1626d55f24a79c4f',
        repo: 'wangxf133.github.io',
        owner: 'wangxf133',
        admin: ['wangxf133'],
        id: location.pathname,      // Ensure uniqueness and length less than 50
        distractionFreeMode: false  // Facebook-like distraction free mode
      })

      gitalk.render('gitalk-container')

    </script>
  

  




  </body>
</html>
